operator-utils-api-service/src/main/java/redhat/ocp4/operators/catalog/utils/cache/OperatorCatalogCache.java

```
@@ -8,98 +8,93 @@
-
-    private static final String IMAGE_OPERATORS_CACHE = "images-to-operators";
-
-    @Value("${operator-catalog.archive.url}")
-    private String archiveDownloadUrl;
-
-    @Value("${operatorhub.tar.gz.file:/opt/operatorhub.tar.gz}")
-    private String operatorHubFilePath;
-
-
-    /**
-     * After creation of the bean, and every five minutes, refresh the Operator Catalog cache. If the file exists
-     * locally, use that file as the source of truth, otherwise download it from Artifactory using the provided URL.
-     */
-    @Scheduled(fixedRateString = "${operator-catalog.cache-refresh-ms}")
-    public void populateCache() {
-        logger.info("Populating operator caches...");
-        try {
-            File operatorHubFile = new File(operatorHubFilePath);
-            InputStream in;
-            if (operatorHubFile.exists()) {
-                logger.debug("Found " + operatorHubFilePath + ", using for local cache.");
-                in = new FileInputStream(operatorHubFile);
-            } else {
-                logger.debug("Did not find " + operatorHubFilePath + ". Attempting download from " + archiveDownloadUrl);
-                in = new URL(archiveDownloadUrl).openStream();
-            }
-
-            Map<String, List<OperatorDetails>> operatorDetailsList = GtarUtil.imageOperatorInfo(in);
-            IMap<String, List<OperatorDetails>> operatorCache = hazelcastInstance.getMap(IMAGE_OPERATORS_CACHE);
-            logger.debug("Clearing existing caches...");
-            operatorCache.clear();
-            logger.debug("Caches cleared!");
-            operatorCache.putAll(operatorDetailsList);
-            logger.info("Successfully populated cache with " + operatorCache.size() + " operators!");
-        } catch (MalformedURLException e) {
-            logger.error("Invalid URL: " + archiveDownloadUrl);
-            e.printStackTrace();
-        } catch (IOException e) {
-            e.printStackTrace();
```

@monkey1016 just thought of something while testing on the cluster.. Here, if there is an IOException problem with populating the cache, we're printing the exception stack trace. I'll add a logger.sever(e.getMessage()) for Kibana to pick up the ERROR.

There are two ways we can handle this if an IOExceptio happens here:
1. as is here, if there is a problem repopulating the cache, the client will just get the previously populated, stale search result
2. we can throw the Exception, bubbling it up to the OperatorCatalogUtilApis api method to handle, and throw a 4XX error back to the client with the error message, letting the client know that there is a problem.

I'm leaning on #2, so that the client will know and have to take some kind of action... Otherwise, its likely a logged error and exception trace will go unnoticed indefinitely in Citi, and clients will just keep hitting stale data. What do you think would make more sense here?

---

I'm good with number 2